src/app/teacher/teacher-run-list/teacher-run-list.component.ts

```diff
@@ -39,19 +41,15 @@ export class TeacherRunListComponent implements OnInit {
   }
 
   private getRuns(): void {
-    forkJoin([this.teacherService.getRuns(), this.teacherService.getSharedRuns()]).subscribe(
-      ([personalRuns, sharedRuns]) => {
-        personalRuns = personalRuns.map((run) => new TeacherRun(run));
-        sharedRuns = sharedRuns.map((run) => {
-          const sharedRun = new TeacherRun(run);
-          sharedRun.shared = true;
-          return sharedRun;
-        });
-        this.runs = personalRuns.concat(sharedRuns);
-        this.processRuns();
-        this.loaded = true;
-      }
-    );
+    this.teacherService.getRuns().subscribe((runs) => {
+      this.runs = runs.map((run) => {
+        const teacherRun = new TeacherRun(run);
+        teacherRun.shared = !teacherRun.isOwner(this.userService.getUserId());
```

Comment:
Should the call to this.userService.getUserId() be outside the runs.map() so that this.userService.getUserId() only gets called once?